src/cargo/core/manifest.rs

```diff
@@ -283,6 +283,8 @@ struct SerializedTarget<'a> {
     doctest: bool,
     /// Whether tests should be run for the target (`test` field in `Cargo.toml`)
     test: bool,
+    /// Whether this should be run with `--test`
+    harness: bool,
```

Comment:
FYI our [contributing docs ask for us to approve Issues before moving on to PRs](https://github.com/rust-lang/cargo/blob/master/CONTRIBUTING.md)